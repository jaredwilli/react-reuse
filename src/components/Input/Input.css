
.mg-input {
    width: 100%;
  }

  .mg-input input {
    border: 1px solid #ccc;
    border-radius: 2px;
    font-size: 15px;
    font-family: BrownStd, arial, sans-serif;
    padding: 12px;
    margin: 5px 0;
    outline: none;
  }
