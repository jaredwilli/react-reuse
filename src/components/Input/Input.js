import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Label from '../Label';

import './Input.css';

/** Input with integrated label to enforce consistency in layout, error display, label placement, and required field marker. */
const Input = ({
    htmlId,
    name,
    placeholder,
    onChange,
    value,
    label,
    children,
    error,
    width,
    height,
    type,
    required,
    ...props
}) => {

    const style = {
        borderColor: error ? 'red' : '#ccc',
        width: width && width,
        height: height && height
    };

    const myClass = classNames('mg-input', {
        ['mg-error']: error
    });

    return (
        <div className={myClass}>
            {label &&
                <Label
                    htmlFor={htmlId}
                    label={label}
                    required={required}
                />
            }

            <input
                id={htmlId}
                type={type}
                name={name}
                placeholder={placeholder}
                value={value}
                onChange={onChange}
                style={style}
                {...props}
            />
            {children}

            {error &&
                <div className={classNames({ 'mg-error-msg': error })}>
                    {error}
                </div>
            }
        </div>
    );
};

Input.defaultProps = {
    type: 'text',
    required: false
};

Input.propTypes = {
    /** Unique HTML ID. Used for tying label to HTML input. Handy hook for automated testing. */
    htmlId: PropTypes.string.isRequired,

    /** Input name. Recommend setting this to match object's property so a single change handler can be used. */
    name: PropTypes.string.isRequired,

    /** Input label */
    label: PropTypes.string,

    /** Input type */
    type: PropTypes.oneOf(['text', 'email', 'number', 'password']),

    /** Mark label with asterisk if set to true */
    required: PropTypes.bool,

    /** Function to call onChange */
    onChange: PropTypes.func,

    /** Placeholder to display when empty */
    placeholder: PropTypes.string,

    /** Width */
    width: PropTypes.string,

    /** Height */
    height: PropTypes.string,

    /** Value */
    value: PropTypes.any,

    /** String to display when error occurs */
    error: PropTypes.string,

    /** Child component to display next to the input */
    children: PropTypes.node
};

export default Input;
