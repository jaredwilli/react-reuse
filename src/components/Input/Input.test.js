import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Input from './Input';

describe('Input', () => {
    const props = {
        htmlId: 'test',
        name: 'Test',
        label: 'Test',
        onChange: () => true
    };

    it('renders the component correctly', () => {
        const tree = renderer.create(<Input {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
