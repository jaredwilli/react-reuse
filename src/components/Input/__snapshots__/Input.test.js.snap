// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders the component correctly 1`] = `
<div
  className="mg-input"
>
  <label
    htmlFor="test"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Test
     
  </label>
  <input
    id="test"
    name="Test"
    onChange={[Function]}
    placeholder={undefined}
    style={
      Object {
        "borderColor": "#ccc",
        "height": undefined,
        "width": undefined,
      }
    }
    type="text"
    value={undefined}
  />
</div>
`;
